### PR TITLE
Match more of HSD LObj

### PR DESCRIPTION
**Pull request by r-burns**
Not finished, but at a point where this is still shiftable.

### CODE REVIEW

File: src/sysdolphin/baselib/lobj.c [hunk 1]
```diff
@@ -220,13 +220,14 @@ void HSD_LObjReqAnimAll(HSD_LObj *lobj, f32 startframe)
     }
 }
 
-Vec const position = {0.0F, 0.0F, 0.0F};
-Vec const interest = {0.0F, 0.0F, 0.0F};
-
-#ifdef NON_MATCHING
+Vec const lbl_803B94A0 = {0.0F, 0.0F, 0.0F};
+Vec const lbl_803B94AC = {0.0F, 0.0F, 0.0F};
```
Comment (ribbanya) [inline src/sysdolphin/baselib/lobj.c:224]:
```suggestion
Vec const lbl_803B94A0 = { 0.0F, 0.0F, 0.0F };
Vec const lbl_803B94AC = { 0.0F, 0.0F, 0.0F };
```

File: src/sysdolphin/baselib/lobj.c [hunk 2]
```diff
@@ -340,106 +285,1065 @@ void HSD_LObjSetup(HSD_LObj *lobj, GXColor color, f32 shininess, u32 unused)
...
+        GXInitLightSpot(&lobj->lightobj, cutoff, spot_func);
+    }
+}
+
+extern char lbl_804D5D24[4];
 #pragma push
-asm void HSD_LObjSetup(HSD_LObj *lobj, GXColor color, f32 shininess, u32 unused)
-{ // clang-format off
+asm void HSD_LObjSetupInit()
+{
```
Comment (ribbanya) [inline src/sysdolphin/baselib/lobj.c:416]: Was this a mistake?

File: src/sysdolphin/baselib/lobj.h [hunk 3]
```diff
@@ -46,34 +46,34 @@
 
 typedef struct _HSD_LightPoint {
     f32 cutoff;
-    u8 point_func;
+    u32 point_func;
     f32 ref_br;
     f32 ref_dist;
-    u8 dist_func;
+    u32 dist_func;
```
Comment (ribbanya) [inline src/sysdolphin/baselib/lobj.h:52]: Are these enums, or pointers? Or unsigned constants?

**Review by ribbanya — CHANGES_REQUESTED** [review]

File: src/sysdolphin/baselib/lobj.h (hunk 3 above)
Comment (r-burns) [reply src/sysdolphin/baselib/lobj.h:52]: Not sure, I don't think their point of use has been decomped yet.

Comment (ribbanya) [reply src/sysdolphin/baselib/lobj.h:52]: I kind of would like there to be a `typedef s32 UnkEnum;` in `types.h` at this point, but what do you think?

File: src/sysdolphin/baselib/lobj.c (hunk 2 above)
Comment (r-burns) [reply src/sysdolphin/baselib/lobj.c:416]: Not sure what you mean?

Comment (ribbanya) [reply src/sysdolphin/baselib/lobj.c:416]:
```diff
-asm void HSD_LObjSetup(HSD_LObj *lobj, GXColor color, f32 shininess, u32 unused)
-{ // clang-format off
+asm void HSD_LObjSetupInit()
+{
```

File: src/sysdolphin/baselib/lobj.h (hunk 3 above)
Comment (r-burns) [reply src/sysdolphin/baselib/lobj.h:52]: I don't think we're certain it's an enum, and hiding the type behind an alias seems to add another layer of indirection for little gain.

Comment (ribbanya) [reply src/sysdolphin/baselib/lobj.h:52]: Hm, alright.

File: src/sysdolphin/baselib/lobj.c (hunk 2 above)
Comment (ribbanya) [reply src/sysdolphin/baselib/lobj.c:416]: Oh, the diff just shifted weirdly. Nevermind.

Comment (r-burns) [reply src/sysdolphin/baselib/lobj.c:416]: HSD_LObjSetup is now decomped. HSD_LObjSetupInit is a different function.

**Review by ribbanya — APPROVED** [review]

**Pull request merged**